### ISSUE TITLE
huggingface/tokenizers: The current process just got forked, after parallelism has already been used.

### ISSUE THREAD

**Issue by milong26** [huggingface/lerobot#1444]
I found this warn when I fine-tune smolvla.
the full output is like:
```
INFO 2025-07-03 14:52:38 ts/train.py:226 Start offline training on a fixed dataset
huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...
To disable this warning, you can either:
        - Avoid using `tokenizers` before the fork if possible
        - Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)
 ```
I solved it by adding this in `train.py` first line.
```
import os
os.environ["TOKENIZERS_PARALLELISM"] = "false"
```
I don't know whether this warn need to be fixed or just passed?
version:
what i modified in train.py
```
# https://github.com/huggingface/lerobot/issues/1377
import os
os.environ["CUDA_VISIBLE_DEVICES"] = "0"
os.environ["TOKENIZERS_PARALLELISM"] = "false"
```
smovla_base: download from git clone https://huggingface.co/lerobot/smolvla_base [Add smolvla tag (](https://huggingface.co/lerobot/smolvla_base/commit/3326b100334ffc0a0bd1ec27e3afb1cfa2a6000c)[#7](https://huggingface.co/lerobot/smolvla_base/discussions/7)[)](https://huggingface.co/lerobot/smolvla_base/commit/3326b100334ffc0a0bd1ec27e3afb1cfa2a6000c)
3326b10
backbone HuggingFaceTB/SmolVLM2-500M-Video-Instruct [Update README.md](https://huggingface.co/HuggingFaceTB/SmolVLM2-500M-Video-Instruct/commit/7b375e1b73b11138ff12fe22c8f2822d8fe03467)
7b375e1
lerobot version: [Add smolvla extra nightly (](https://github.com/huggingface/lerobot/commit/483be9aac217c2d8ef16982490f22b2ad091ab46)https://github.com/huggingface/lerobot/pull/1408[)](https://github.com/huggingface/lerobot/commit/483be9aac217c2d8ef16982490f22b2ad091ab46) 483be9aac217c2d8ef16982490f22b2ad091ab46

**Comment by Viredery** [comment]
Same here.

**Comment by milong26** [comment]
@Viredery hi how long your training take for 200 step? It takes me around 40min... But I only use 50 episode(640*480image,2 camera,and 15 second each episode) and fine-tune 450m smolvla_base. 
And in the log, it said num_learnablr_params 100M,num_total_params 450M, it shouldn't take so much time on gpu

**Comment by Viredery** [comment]
> [@Viredery](https://github.com/Viredery) hi how long your training take for 200 step? It takes me around 40min... But I only use 50 episode(640*480image,2 camera,and 15 second each episode) and fine-tune 450m smolvla_base. And in the log, it said num_learnablr_params 100M,num_total_params 450M, it shouldn't take so much time on gpu

For me, it takes 12min for 200 steps and the batch size is 256

**Comment by milong26** [comment]
@Viredery Hi! Have you tested your fine-tuned model? I tested but got very bad performance, considering where makes it such bad ;-;

**Comment by Viredery** [comment]
> [@Viredery](https://github.com/Viredery) Hi! Have you tested your fine-tuned model? I tested but got very bad performance, considering where makes it such bad ;-;

Haven't tested yet, Because the loss in experiment is not low enough (around 0.03), much higher than other models, consider the same dataset

**Comment by milong26** [comment]
@Viredery Thanks! By the way I found the code of smolvla is not completed. https://github.com/huggingface/lerobot/issues/1251 from this I heard that Async Inference is still merging now.
 
Async inference merged.
